lintro/ai/review/context/collection.py

```
@@ -0,0 +1,561 @@
+            return file_path.read_text(encoding="utf-8", errors="surrogateescape")
+
+        result = _run_git(args=["show", f"{head_ref}:{path}"], check=False)
+        if result.returncode == 0 and result.stdout:
+            return result.stdout
+        if repo is not None:
+            return _read_workflow_post_image_via_gh(
+                path=path,
+                head_ref=head_ref,
+                repo=repo,
+            )
+        return None
```

Ruff (0.15.18)

[warning] Consider moving this statement to an `else` block

(TRY300)